src/plugin/file/kml/kmlsource.js

```diff
@@ -740,3 +741,72 @@ plugin.file.kml.KMLSource.prototype.restore = function(config) {
     this.setMinRefreshPeriod(config['minRefreshPeriod']);
   }
 };
+
+
+/**
+ * Creates a basic feature overlay used to animate features on the map.
+ * @protected
+ * @override
+ */
+plugin.file.kml.KMLSource.prototype.createAnimationOverlay = function() {
+  plugin.file.kml.KMLSource.superClass_.createAnimationOverlay.call(this);
+
+  // Hide all annotations
+  for (var i = 0, n = this.getFeatures().length; i < n; i++) {
+    var feature = this.getFeatures()[i];
+    var annotation = this.getFeatureNode(feature).getAnnotation();
+    if (annotation) {
+      annotation.setVisible(false);
+    }
+  }
+};
+
+
+/**
+ * Updates features displayed by the animation overlay if it exists.
+ * @protected
+ * @override
+ */
+plugin.file.kml.KMLSource.prototype.updateAnimationOverlay = function() {
+  if (this.animationOverlay) {
+    // hide annotations in overlay
+    for (var i = 0, n = this.animationOverlay.getFeatures().length; i < n; i++) {
+      var feature = this.animationOverlay.getFeatures()[i];
+      var annotation = this.getFeatureNode(feature).getAnnotation();
+      if (annotation) {
+        annotation.setVisible(false);
+      }
+    }
+
+    plugin.file.kml.KMLSource.superClass_.updateAnimationOverlay.call(this);
```

Comment:
Please split this function into two separate `if`'s with the parent call in between, so the parent function is always called.

While it wouldn't currently impact execution, we should defensively code to handle changes to `os.source.Vector#updateAnimationOverlay` that do something when the animation overlay is not defined.